### PR TITLE
feat: Symbolic based Symbolication

### PR DESCRIPTION
**Pull request by mitsuhiko**
This will eventually replace symsynd. Right now it's broken mess still

General TODOs:

* [x] ensure we delete symcache files upon dsym deletion / org deletion
* [x] verify the on file system cache pruning works
* [x] prefetch blobs from s3 in one go. concurrency?
* [x] can we avoid in-memory and go to fs in a few places?
* [x] release symbolic update to pypi

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR includes migrations</td>
    </tr>
  </tbody>
</table>

## Migration Checklist

- [ ] new columns need to be nullable (unless table is new)
- [ ] migration with any new index needs to be done concurrently
- [ ] data migrations should not be done inside a transaction
- [ ] before merging, check to make sure there aren't conflicting migration ids


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Comment by mitsuhiko** [comment]
This now includes a squash of this: https://github.com/getsentry/sentry/pull/6279

**Comment by mitsuhiko** [comment]
Wheels for testing:

* [symbolic-0.0.1-py2.py3-none-macosx_10_10_intel.whl](https://s3.amazonaws.com/getsentry-builds/getsentry/symbolic/9637ae816febe91597dce50a3e23a2acc68e4679/symbolic-0.0.1-py2.py3-none-macosx_10_10_intel.whl)
* [symbolic-0.0.1-py2.py3-none-manylinux1_i686.whl](https://s3.amazonaws.com/getsentry-builds/getsentry/symbolic/9637ae816febe91597dce50a3e23a2acc68e4679/symbolic-0.0.1-py2.py3-none-manylinux1_i686.whl)
* [symbolic-0.0.1-py2.py3-none-manylinux1_x86_64.whl](https://s3.amazonaws.com/getsentry-builds/getsentry/symbolic/9637ae816febe91597dce50a3e23a2acc68e4679/symbolic-0.0.1-py2.py3-none-manylinux1_x86_64.whl)

**Review by Adam8989 — APPROVED** [review]

**Comment by mitsuhiko** [comment]
I think this now works. I want to port some more tests and we need an update for symbolic to pypi.

**Comment by mitsuhiko** [comment]
This should be ready for review now. Coverage is better than before but still not great.

**Comment by mitsuhiko** [comment]
We found some regressions on really weird symbol files we want to look at, but for a test deployment I think we're good to go.

File: src/sentry/models/dsymfile.py [hunk 1]
```diff
@@ -183,11 +189,37 @@ def dsym_type(self):
...
+    def supports_symcache(self):
+        # Only one that supports it so far.
+        return self.dsym_type == 'macho'
+
     def delete(self, *args, **kwargs):
         super(ProjectDSymFile, self).delete(*args, **kwargs)
         self.file.delete()
 
 
+class ProjectSymCacheFile(Model):
```
Comment (mattrobenolt) [inline src/sentry/models/dsymfile.py]: This model is fine, but we need to add it to all the deletion stuff properly.

Comment (mitsuhiko) [reply src/sentry/models/dsymfile.py]: Is there more deletion code? I added it to the place I found.

Comment (mattrobenolt) [reply src/sentry/models/dsymfile.py]: Oh, you're right. All good.

Let's jsut get this model pulled out into it's own PR.

**Comment by mitsuhiko** [comment]
The migration went to a different PR and pushing out a new symbolic build now that will fix travis.

**Comment by mitsuhiko** [comment]
I opened a squashed PR we can try to deploy to a single worker.

**Comment by mitsuhiko** [comment]
Closing this for the squashed one.

**Pull request closed without merging**